pycbc/vetoes/sgchisq.py

```diff
@@ -0,0 +1,97 @@
+from pycbc.waveform.utils import apply_fseries_time_shift
+from pycbc.filter import sigma            
+from pycbc.waveform import sinegauss
+from pycbc.vetoes.chisq import SingleDetPowerChisq
+import numpy
+
+class SingleDetSGChisq(SingleDetPowerChisq):
+    """Class that handles precomputation and memory management for efficiently
+    running the power chisq in a single detector inspiral analysis.
+    """
+    def __init__(self, bank, num_bins=0,
+                       snr_threshold=None,
+                       chisq_locations=None):
+        if snr_threshold:
+            self.do = True
+            self.num_bins = num_bins
+        else:
+            self.do = False
+        
+        self.snr_threshold = snr_threshold
+        self.params = {}
+        for descr in chisq_locations:
+            region, values = descr.split(":")
+            mask = bank.table.parse_boolargs([(1, region), (0, 'else')])[0]
+            hashes = bank.table['template_hash'][mask.astype(bool)]
+            for h in hashes:
+                self.params[h] = values
+
+    def values(self, stilde, template, psd, snrv, snr_norm, indices):
+        """ Calculate sine-Gaussian chisq
+
+        Returns
+        -------
+        chisq: Array
+            Chisq values, one for each sample index
+        """
+        if template.params.template_hash not in self.params:
+            return numpy.ones(len(snrv))
+        
+        values = self.params[template.params.template_hash].split(',')
+        if not self.do:
+            return None
+        
+        # Get the chisq bins to use as the frequency reference point
```

Comment:
not clear.  comment could say something about why we need a frequency reference point